File: firstProject/src/main/resources/static/app/pages/main/notice/notice.service.js
(function(){
	'use strict';
	
	angular.module('myApp.main.notice')
		.service('noticeService', noticeService)
	
	function noticeService($http, config){
		console.info("noticeService");
		
		this.getNoticeList = function getNoticeList(param){
			return $http.get(config.apiUrl + 'notice/getNoticeList', {params: param}).then(function(response){
				return response;
			});
		};
		 
		this.getNoticeDetail = function getNoticeDetail(param){
			return $http.get(config.apiUrl + 'notice/getNoticeDetail', {params: param}).then(function(response){
				return response;
			});
		};
		
		this.noticeInsert = function noticeInsert(param){
			return $http.post(config.apiUrl + 'notice/noticeInsert', param).then(function(response){
				return response;
			});
		};
		
		this.noticeUpdate = function noticeUpdate(param){
			return $http.post(config.apiUrl + 'notice/noticeUpdate', param).then(function(response){
				return response;
			});
		};
		
	};

})();

File: firstProject/src/main/java/com/project/firstProject/notice/service/NoticeService.java
package com.project.firstProject.notice.service;

import java.util.List;

import com.project.firstProject.notice.vo.NoticeVO;

public interface NoticeService {

	public List<NoticeVO> getNoticeList(NoticeVO noticeVo);
	
	public int getNoticeCnt(NoticeVO noticeVo);
	
	public NoticeVO getNoticeDetail(NoticeVO noticeVo);
	
	public int noticeInsert(NoticeVO noticeVo);
	
	public int noticeUpdate(NoticeVO noticeVo);
	
	public int noticeDelete(NoticeVO noticeVo);
	
}
